src/nix/flake.md

```diff
@@ -364,6 +364,33 @@ Currently the `type` attribute can be one of the following:
   * `sourcehut:~misterio/nix-colors/182b4b8709b8ffe4e9774a4c5d6877bf6bb9a21c`
   * `sourcehut:~misterio/nix-colors/21c1a380a6915d890d408e9f22203436a35bb2de?host=hg.sr.ht`
 
+* `gitea`: Similar to `github`, is a more efficient way to fetch
+  Gitea/Forgejo repositories. The default host is `codeberg.org`.
+  The following attributes are required:
+
+  * `owner`: The owner of the repository.
+
+  * `repo`: The name of the repository.
+
+  Like `github`, these are downloaded as tarball archives.
+
+  The URL syntax for `gitea` flakes is:
+
+  `gitea:<owner>/<repo>(/<rev-or-ref>)?(\?<params>)?`
+
+  `<rev-or-ref>` works the same as `github`. Either a branch or tag name
+  (`ref`), or a commit hash (`rev`) can be specified.
+
+  Since Gitea/Forgejo allows for self-hosting, you can specify `host` as
+  a parameter, to point to any instances other than `codeberg.org`.
+
+  Some examples:
+
+  * `gitea:redict/redict`
+  * `gitea:redict/redict/main`
+  * `gitea:redict/redict/a4c81102327bc2c74d229784a1d1dd680c708918`
+  * `gitea:lix-project/lix?host=git.lix.systems`
```

Comment:
Maybe we should also support this?

* `gitea:git.lix.systems/some-subdirectory/lix-project/lix`

Reply:
I agree that we should support sub-directory. Though, I'd like to keep the `host` argument, so the URL doesn't feel to far off compared to the existing `github:` and `sourcehut:` URL types.

Reply:
>     * `gitea:git.lix.systems/some-subdirectory/lix-project/lix`
> 
> 
> Also some-subdirectory could become a url parameter

What do you mean by subdirectories?

Subdirectories in the gitea repository are already supported with the `dir` url parameter.

Reply:
> I wonder if we actually should default to codeberg and not just make the host part of the url? This doesn't look much more complex and promotes self-hosting (which is in the spirit of gitea):

I don't think we should actually default to codeberg becuase. I just use codeberg and wanted to quickly simplify my flakes.

I currently have an implementation which requires the specification of the host in the url in the [`gitea-fetcher-explicit-url branch`](https://github.com/tilmanmixyz/nix/blob/gitea-fetcher-explicit-url/src/libfetchers/github.cc#L619) of my fork.
But currently it makes the `host` parameter useless since it requires it in the scheme.